Assets/Scripts/UI/LevelSelect.cs

```diff
@@ -86,15 +102,63 @@ private void LazyLoad()
             FindObjectOfType<SaveDataManager>()?.PreloadSaveData();
         }
 
-        private void OnLevelSelected(string sceneName, Button clickedButton)
+        // <summary>
+        // retrieves the name of the level from the cutscene name
+        // sceneName must be one of: "IntroCutscene" "SecondCutscene" "ThirdCutscene" "FourthCutscene"
+        // <summary>
+        private string GetLevelName(string sceneName)
+        {
+            if (!allLevelLoadingSceneNames.Contains(sceneName))
+            {
+                return "";
+            }
+            int idx = allLevelLoadingSceneNames.IndexOf(sceneName);
```

Comment:
same comment about how IndexOf returns -1 if it's not found so it can be simplified